cluster-autoscaler/cloudprovider/aws/aws_wrapper_test.go

```diff
@@ -695,3 +695,42 @@ func TestBuildLaunchTemplateFromSpec(t *testing.T) {
 		assert.Equal(unit.exp, got)
 	}
 }
+
+func TestGetInstanceTypesFromInstanceRequirementsWithEmptyList(t *testing.T) {
+	e := &ec2Mock{}
+	awsWrapper := &awsWrapper{
+		autoScalingI: nil,
+		ec2I:         e,
+		eksI:         nil,
+	}
+	requirements := &ec2.InstanceRequirementsRequest{}
+
+	e.On("DescribeImages", &ec2.DescribeImagesInput{
+		ImageIds: []*string{aws.String("123")},
+	}).Return(&ec2.DescribeImagesOutput{
+		Images: []*ec2.Image{
+			{
+				Architecture:       aws.String("x86_64"),
+				VirtualizationType: aws.String("xen"),
+			},
+		},
+	})
+	e.On("GetInstanceTypesFromInstanceRequirementsPages",
+		&ec2.GetInstanceTypesFromInstanceRequirementsInput{
+			ArchitectureTypes:    []*string{aws.String("x86_64")},
+			InstanceRequirements: requirements,
+			VirtualizationTypes:  []*string{aws.String("xen")},
+		},
+		mock.AnythingOfType("func(*ec2.GetInstanceTypesFromInstanceRequirementsOutput, bool) bool"),
+	).Run(func(args mock.Arguments) {
+		fn := args.Get(1).(func(*ec2.GetInstanceTypesFromInstanceRequirementsOutput, bool) bool)
+		fn(&ec2.GetInstanceTypesFromInstanceRequirementsOutput{
+			InstanceTypes: []*ec2.InstanceTypeInfoFromInstanceRequirements{},
+		}, false)
+	}).Return(nil)
+
+	result, err := awsWrapper.getInstanceTypeFromInstanceRequirements("123", requirements)
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), "unable to get instance types from requirements")
```

Comment:
Can we use `assert.ErrorIs` here instead of `Contains`?

Reply:
I just found `assert.EqualError` is more commonly used in the codebase :P